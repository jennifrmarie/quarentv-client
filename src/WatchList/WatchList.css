.WatchedListNav {
    display: flex;
    width: 33%;
}