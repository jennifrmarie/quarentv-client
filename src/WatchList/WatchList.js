import React, { Component } from 'react'
import AppContext from '../AppContext'
import moment from 'moment';

export default class WatchList extends Component {
    static contextType = AppContext
    render() {

        return (
            <div className="WatchedListNav">
                <h2>What you've watched</h2>
                <ul className='WatchedList_list'>
                    {this.context.watched.map(entry =>
                        <li className='watchlist_main' key={entry.id}>
                            {entry.title}{' '}
                            {entry.strmservice}{' '}
                            {moment(entry.date).format('MM-DD-YYYY')}
                            <div>
                                <label htmlFor="unwatch">unwatch</label>
                                <input type="checkbox" onClick={e => this.props.handleUnwatchButton(entry.id)} name='checkbox' id='checkbox' />
                            </div>
                        </li>)}
                </ul>
            </div>
        )
    }
}
