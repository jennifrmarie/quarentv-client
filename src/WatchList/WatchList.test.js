import React from 'react';
import ReactDOM from 'react-dom';
import WatchList from './WatchList';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<WatchList />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
    const tree = renderer
      .create(<WatchList name="Messages" unread={4}/>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
    });