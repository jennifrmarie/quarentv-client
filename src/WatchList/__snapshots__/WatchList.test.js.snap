// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
<div
  className="WatchedListNav"
>
  <h2>
    What you've watched
  </h2>
  <ul
    className="WatchedList_list"
  />
</div>
`;
